s3-discovery/handler.py

```diff
@@ -0,0 +1,57 @@
+import boto3
+
+s3 = boto3.resource(
+    "s3",
+)
+
+
+def list_bucket(bucket, prefix, file_type):
+    try:
+        files = []
+        bucket = s3.Bucket(bucket)
+        for obj in bucket.objects.filter(Prefix=prefix):
+            if file_type:
+                if obj.key.endswith(file_type):
+                    files.append(obj.key)
+            else:
+                files.append(obj.key)
+        return files
+
+    except Exception as e:
+        print(e)
+    return e
+
+
+def handler(event, context):
+    filenames = list_bucket(
+        bucket=event["bucket"], prefix=event["prefix"], file_type=event["file_type"]
+    )
+
+    files_objs = []
+    for f in filenames:
+        files_objs.append(
+            {
+                # Remove trailing back slash used for prefixing
+                "collection": event["prefix"][:-1],
+                "s3_filename": f's3://{event["bucket"]}/{f}',
+                "datetime_regex": {
+                    "regex": "^(.*?)(_)([0-9][0-9][0-9][0-9])(.tif)$",
```

Comment:
Is this regex only going to work for files that end in `.tif`? there may be a case in the future where we use this to discover files of another type in S3 for downstream processing.

Reply:
good point, i updated the regex to depend on the filetype. realistically, we would will want / need to pass the regex to the discovery task as well if its chained to the stac gen task.